src/Behat/Testwork/Cli/ConvertConfigCommand.php

```diff
@@ -0,0 +1,115 @@
+    private OutputInterface $output;
+
+    public function __construct(
+        private ConfigurationLoader $configurationLoader,
+        ExtensionManager $extensionManager
+    ) {
+        parent::__construct('convert-config');
+        Profile::setExtensioManager($extensionManager);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->output = $output;
+        $configPath = $this->configurationLoader->getConfigurationFilePath();
+        if (str_ends_with($configPath, '.php')) {
+            throw new ConfigurationLoadingException(sprintf('Configuration file `%s` is already in PHP format', $configPath));
+        }
+
+        $this->output->writeln(['', 'Starting conversion']);
+        $this->convertFile($configPath);
+        $this->output->writeln(['Conversion finished', '']);
+        return 0;
+    }
+
+    private function convertFile(string $filePath): string
+    {
+        $this->output->writeln('Converting configuration file: ' . $filePath);
+
+        $yamlConfig = (array) Yaml::parse(file_get_contents($filePath));
+
+        if (isset($yamlConfig[Config::IMPORTS_SETTING])) {
+            $imports = $yamlConfig[Config::IMPORTS_SETTING];
+            $basePath = rtrim(dirname($filePath), DIRECTORY_SEPARATOR)  . DIRECTORY_SEPARATOR;
+
+            $newImports = [];
+            foreach ($imports as $import) {
+                $import = $basePath . $import;
+                $newImports[] = substr($this->convertFile($import), strlen($basePath));
+            }
+            $yamlConfig[Config::IMPORTS_SETTING] = $newImports;
+        }
+
+        $config = new Config($yamlConfig);
+
+        $factory = new BuilderFactory();
+
+        $expr = $config->toPhpExpr();
+
+        $configStms = [new Return_($expr)];
+
+        $traverser = new NodeTraverser();
+        $useStatementAdder = new UseStatementAdder();
+        $traverser->addVisitor($useStatementAdder);
+        $configStms = $traverser->traverse($configStms);
+
+        $useStmts = [];
+
+        $useStatements = $useStatementAdder->getUseStatements();
+        sort($useStatements);
+        foreach ($useStatements as $useStatement) {
+            $useStmts[] = $factory->use($useStatement)->getNode();
+        }
+        if (count($useStmts) !== 0) {
+            $useStmts[] = new Nop();
+        }
+
+        $stmts = array_merge($useStmts, $configStms);
+        $stmts[] = new Nop();
+
+        // Find the name of the output file
+        $outputFileName = $filePath;
+        // First, handle cases where `.dist` is part of the extension
+        $outputFileName = preg_replace('/(\.ya?ml)\.dist$|\.dist\.(ya?ml)$/', '.dist.php', $outputFileName);
+        // Then, handle regular `.yaml` or `.yml` files
+        $outputFileName = preg_replace('/\.(ya?ml)$/', '.php', $outputFileName);
+
+        $printer = new CustomPrettyPrinter();
+
+        file_put_contents($outputFileName, $printer->prettyPrintFile($stmts));
+
```

Comment:
Would it be worth / would it be hard to re-read the generated PHP config, call the `$config->toArray()` and assert that the result is identical to the values from the YAML, as a final sanity/safety check that there aren't any odd userland edge cases that we've missed and accidentally removed from their files?

Reply:
I tried to implement this but found out that there are some cases where we implement default values differently, so for example if you have a yaml file with just `default:`, this will be transformed into an array `['default' => null]` when reading the yaml but for the php config we end up with `['default' => []]`. The final functionality is the same but trying to deal with these differences when checking the conversion is complex so I decided not to add this check